### ISSUE TITLE
Kessel Run should still be able to be played when another copy is stacked on code clearance

### ISSUE THREAD

**Issue by jdipaolo1** [PlayersCommittee/gemp-swccg-public#10]

**Comment by BOTVHD** [comment]
Is (or was?) this reproducible only for Kessel Run?  Or was it for all utinni effects that retrieve? Or for all unique cards that Code Clearance grabs?

Did this happen with the "Do They Have A Code Clearance?" effect or the "Do They Have A Code Clearance?" defensive shield?

Stacked cards on Code Clearance should be in a supporting state.

EDIT: Found the original thread that (I assume) the issue was ported in from: https://forum.starwarsccg.org/viewtopic.php?t=74300 

In summary, it was regular Kessel Run (non-V) and the Code Clearance defensive shield.

<img width="956" height="495" alt="Image" src="https://github.com/user-attachments/assets/b8d1705d-fb5f-4f39-bea5-a740a4be98e4" />

There was one other (slightly older) report of this issue happening, too (but replay is not accessible): https://forum.starwarsccg.org/viewtopic.php?p=1173284#p1173284

**Comment by BOTVHD** [comment]
I tried to reproduce this with a simpler test rig case (play second copy of Kessel Run in hand after completing one and having it grabbed by the shield).  It worked correctly, so it seems like one of the following:
- this has been fixed
- intermittent/pseudo-random
- there is some card interaction in the more complex situation from the screenshot that is not part of the test

I suspect this is a fixed issue (maybe it was originally seeing the stacked card as inactive instead of supporting?), but there are so many places where a fix could have happened, that I haven't found a "smoking gun" change that has obviously fixed this.

**Comment by BOTVHD** [comment]
Smoking gun fix found...

I believe this was fixed on Jun 10, 2021 with https://github.com/PlayersCommittee/gemp-swccg/pull/910 (private repo) as part of several bug fixes that included this description:

> Kessel Run should not be inactive when being grabbed by Code Clearance

Change was in StackCardsFromTableSimultaneouslyEffect.java:
`                                game.getGameState().stackCard(card, _stackOn, _faceDown, !_faceDown && (card.getPreviousCardState() == CardState.ACTIVE || card.getPreviousCardState() == CardState.INACTIVE), false);`

to:

`                                game.getGameState().stackCard(card, _stackOn, _faceDown, !_faceDown && (card.getPreviousCardState() == CardState.ACTIVE || card.getPreviousCardState() == CardState.INACTIVE) && !Filters.grabber.accepts(game, _stackOn), false);`

Along with the test I recently made that confirms correct behavior, I think this can be safely closed, @Gergall ?

**Comment by Gergall** [comment]
I can confirm in my testing today that Kessel Run is able to be deployed while another Kessel Run is stacked on Code Clearance.

Combined with VHD's detailed review and findings and tests (thank you VHD!) I am quite confident that all is good here. Closing this.
